Stack/Opc.Ua.Core/Stack/Tcp/TcpMessageSocket.cs

```diff
@@ -776,29 +708,22 @@ private void OnSocketConnected(object sender, SocketAsyncEventArgs args)
         {
             var socket = sender as Socket;
             bool success = false;
+
             lock (m_socketLock)
             {
-                m_socketResponses--;
                 if (!m_closed && m_socket == null)
                 {
                     if (args.SocketError == SocketError.Success)
                     {
                         m_socket = socket;
                         success = true;
-                        m_tcs.SetResult(args.SocketError);
-                    }
-                    else if (m_socketResponses == 0)
-                    {
-                        m_tcs.SetResult(args.SocketError);
                     }
                 }
             }
 
-            if (success)
-            {
-                ((CallbackAction)args.UserToken)(args.SocketError);
-            }
-            else
+            ((CallbackAction)args.UserToken)(args.SocketError);
```

Comment:
yes, previously the error case was handled in the return code path of BeginConnect, because both IPv4 and IPv6 connection were able to connect concurrently. Now it can directly call back.